picard/coverart/__init__.py

```diff
@@ -164,6 +164,26 @@ def next_in_queue(self):
             self.next_in_queue()
             return
 
+        # local files
+        try:
+            if coverartimage.filepath:
+                data = None
+                try:
+                    with open(coverartimage.filepath, 'rb') as file:
+                        self._set_metadata(coverartimage, file.read())
+                except IOError, (errnum, errmsg):
+                    log.error("Failed to read %r: %s (%d)" %
+                              (coverartimage.from_file, errmsg, errnum))
+                except CoverArtImageIOError:
+                     # It doesn't make sense to store/download more images if we can't
+                     # save them in the temporary folder, abort.
+                     return
+                self.next_in_queue()
+                return
+        except AttributeError:
```

Comment:
Is this for handling the missing `filepath` attribute on images that don't have it? If so, that seems overly broad and should be replaced with `isinstance` or `hasattr` calls.